src/main/java/mops/businesslogic/prometheus/PrometheusComponent.java

```diff
@@ -0,0 +1,147 @@
+package mops.businesslogic.prometheus;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import lombok.extern.slf4j.Slf4j;
+import mops.businesslogic.DirectoryService;
+import mops.businesslogic.FileInfoService;
+import mops.businesslogic.Group;
+import mops.businesslogic.GroupService;
+import mops.exception.MopsException;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Slf4j
+@Component
+public class PrometheusComponent {
+
+    /**
+     * Group service.
+     */
+    private final transient GroupService groupService;
+    /**
+     * File info service.
+     */
+    private final transient FileInfoService fileInfoService;
+    /**
+     * Directory service.
+     */
+    private final transient DirectoryService directoryService;
+    /**
+     * Meter registry.
+     */
+    private final transient MeterRegistry meterRegistry;
+
+    /**
+     * Constructor.
+     *
+     * @param groupService     group service
+     * @param fileInfoService  file info service
+     * @param directoryService directory service
+     * @param meterRegistry    meter registry
+     */
+    public PrometheusComponent(GroupService groupService, FileInfoService fileInfoService,
+                               DirectoryService directoryService, MeterRegistry meterRegistry) {
+        this.groupService = groupService;
+        this.fileInfoService = fileInfoService;
+        this.directoryService = directoryService;
+        this.meterRegistry = meterRegistry;
+
+        addTotalGauges();
+        addGroupGauges();
+    }
+
+    @SuppressWarnings("PMD.LawOfDemeter") //This is a builder
+    private void addTotalGauges() {
+        log.debug("Adding total storage gauge.");
+        Gauge
+                .builder("mops.material1.totalStorageUsage", () -> {
+                            try {
+                                return fileInfoService.getStorageUsage();
+                            } catch (MopsException ignored) {
+                                return 0L;
```

Comment:
Shouldn't we log this? I'm mean for the stacktrace?

Reply:
ah yeah
there is log.cathing(e) I think